core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -180,32 +180,36 @@ void* HIPManagedSpace::impl_allocate(
   if (arg_alloc_size > 0) {
     if (is_first_hip_managed_allocation.exchange(false) &&
         Kokkos::show_warnings()) {
-      if (!hip_driver_check_page_migration(m_device)) {
-        std::cerr << R"warning(
+      do {  // hack to avoid spamming users with too many warnings
+        if (!hip_driver_check_page_migration(m_device)) {
+          std::cerr << R"warning(
 Kokkos::HIP::allocation WARNING: The combination of device and system configuration
                                  does not support page migration between device and host.
                                  HIPManagedSpace might not work as expected.
                                  Please refer to the ROCm documentation on unified/managed memory.)warning"
-                  << std::endl;
-      }
-
-      // check for correct runtime environment
-      const char* hsa_xnack = std::getenv("HSA_XNACK");
-      if (!hsa_xnack)
-        std::cerr << R"warning(
+                    << std::endl;
+          break;  // do not warn about HSA_XNACK environement variable
+        }
+
+        // check for correct runtime environment
+        const char* hsa_xnack = std::getenv("HSA_XNACK");
+        if (!hsa_xnack)
+          std::cerr << R"warning(
```

Comment:
why the do-while hack and not just changing the `if` to an `else if` here and move the `getenv` to the top? I think it would be easier to read

Reply:
I didn't want to have a 4-level nested if-statement which I find arguably harder to read.
Let me know if you want me to change.

Reply:
I leave this to your judgement

Reply:
@Rombur Your call.  Do you want to change or keep as is?